code/modules/modular_computers/file_system/programs/ntnrc_client.dm

```diff
@@ -267,3 +267,58 @@
 		data["messages"] = list()
 
 	return data
+
+// SKYRAT EDIT ADDITION BEGIN
+/datum/computer_file/program/chatclient/proc/reject_bad_message(text)
+	var/non_whitespace = FALSE
+	var/char = ""
+	for(var/i = 1, i <= length(text), i += length(char))
+		char = text[i]
+		switch(text2ascii(char))
+			if(0 to 31)
+				return
+			if(32)
+				continue
+			else
+				non_whitespace = TRUE
+		if (non_whitespace)
+			return text
+
+/datum/computer_file/program/chatclient/proc/reject_bad_username(text, max_length = 32)
+	var/non_whitespace = FALSE
+	var/char = ""
+	
+	if(length(text) > max_length)
+		return
+	else
+		for(var/i = 1, i <= length(text), i += length(char))
+			char = text[i]
+			switch(text2ascii(char))
+				if(0 to 31)
+					return
+				if(32)
+					continue
+				else
+					non_whitespace = TRUE
+		if (non_whitespace)
+			return text
+	
+/datum/computer_file/program/chatclient/proc/reject_bad_chatname(text, max_length = 12)
+	var/non_whitespace = FALSE
+	var/char = ""
+	
+	if (length(text) > max_length)
+		return
+	else
+		for(var/i = 1, i <= length(text), i += length(char))
+			char = text[i]
+			switch(text2ascii(char))
+				if(0 to 31)
+					return
+				if(32)
+					continue
+				else
+					non_whitespace = TRUE
+		if (non_whitespace)
+			return text
+//SKYRAT EDIT ADDITION END
```

Comment:
This should probably be put in another file in our `modular_skyrat` folder, they don't absolutely need to be there.

Reply:
Okay, I'll redo it soon.

Reply:
I put the procedures in a separate module on your advice. Should I put the [MODULAR] tag in the name of the PR?